std/protocols/lookup.asm

```diff
@@ -102,11 +103,10 @@ let lookup: Constr -> () = constr |lookup_constraint| {
 
     let is_first: col = std::well_known::is_first;
 
-    let (acc_1, acc_2) = unpack_ext(acc_ext);
     // First and last acc needs to be 0
     // (because of wrapping, the acc[0] and acc[N] are the same)
-    is_first * acc_1 = 0;
-    is_first * acc_2 = 0;
+    is_first * (acc[0] - 1) = 0;
+    array::new(array::len(acc) - 1, |i| is_first * acc[i + 1] = 0);
```

Comment:
What I understand from this is that on the first row:
- first acc must be 1
- other accs must be 0

I can't see how that was the behavior before (both zero?) nor how it fits the above comment

Could you explain this?

Reply:
Oh, funny, good catch! It's a copy-paste error and comes from the permutation implementation, where we multiply so we want the acc to start with one.

I wondered how this didn't fail at least in the mock prover, but I think it's because the initial value actually doesn't matter: Because of wrapping, the first and last acc are constrained to be equal. It doesn't really matter which value you start with, as long as you end up with the same.

This is different in the permutation case, where we do want to make sure we don't start with 0 (as the entire product would be 0, regardless of the values in the permutation).

Will change it anyway for clarity.

Reply:
Makes sense! Does this mean we could save some constraints here? What if we removed them entirely?